src/io/evented.cr

```diff
@@ -30,22 +30,18 @@ module IO::Evented
     resume_pending_readers
   end
 
-  def evented_write(slice : Bytes, errno_msg : String, &) : Nil
-    return if slice.empty?
-
+  def evented_write(slice : Bytes, errno_msg : String, &) : Int32
     begin
       loop do
-        # TODO: Investigate why the .to_i64 is needed as a workaround for #8230
-        bytes_written = (yield slice).to_i64
```

Comment:
**Question**: we don't need the i64 workaround for #8230?

Reply:
The issue seems to be related to the iteration which advances `slice` until it is empty. After this refactor, this precondition has vanished because the iteration is moved out of here. The bug does not reproduce anymore.

It also doesn't reproduce with the latest Crystal release if we remove this workaround. Still unclear what exactly was causing, but it seems to be fixed and we don't need the workaround anymore whatsoever.

Reply:
I suppose we could make a separate PR to remove the workaround first before refactoring, for good measure 🤷

Reply:
Nice! No need to extract. LGTM!